### PR TITLE
*: add cbo switch.

### PR DESCRIPTION
**Pull request by hanfei1991**
rc4 release will use new planner by default. I add a session variable to control it.

### CODE REVIEW

**Comment by hanfei1991** [comment]
@shenli PTAL

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -90,6 +90,9 @@ const (
 	// It controls the max row count of outer table when do index nested loop join without hint.
 	// After the row count of the inner table is accurate, this variable will be removed.
 	TiDBMaxRowCountForINLJ = "tidb_max_row_count_for_inlj"
+
+	// tidb_cbo will use new planner with cost based optimizer.
```
Comment (zimulala) [inline sessionctx/variable/tidb_vars.go]: s/will use/uses

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func (c *CopClient) IsRequestTypeSupported(reqType, subType int64) bool {
 			return supportExpr(tipb.ExprType(subType))
 		}
 	case kv.ReqTypeDAG:
-		return c.store.mock
```
Comment (zimulala) [inline store/tikv/coprocessor.go]: Can we remove the variable of `mock`?

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**